testing/forge-script/main.star

```diff
@@ -41,52 +42,59 @@ def deploy_contracts(plan, deployment):
     else:
         contract_path = SOURCE_DIR_PATH
 
-    # Check if dependency status is set to true
-    if dependency_status:
+    if dependency_type == "local":
         # Run shell script
         plan.exec(
             service_name = foundry_service.name,
             recipe = ExecRecipe(
                 command = ["/bin/sh", "-c", "sh {}/{}".format(DEPENDENCY_DIR_PATH, dependency_path)],
             ),
         )
-
-    result = plan.exec(
-        service_name = foundry_service.name,
-        recipe = ExecRecipe(
-            command = ["/bin/sh", "-c", "cd {} && forge build".format(contract_path)],
-        ),
-    )
-    plan.verify(result["code"], "==", 0)
-
-    script_output = exec_on_service(
-        plan,
-        foundry_service.name,
-        "cd {} && forge script {}:{} --broadcast --rpc-url {} {} --json  --skip test > output.json ".format(
-            contract_path,
-            script_path,
-            contract_name,
-            rpc_url,
-            wallet_command,
-        ),
-    )
+    elif dependency_type == "git":
+        plan.exec(
+            service_name = foundry_service.name,
+            recipe = ExecRecipe(
+                command = ["/bin/sh", "-c", "cd {} && sh {}".format(contract_path, dependency_path)],
+            ),
+        )
+    if script_path:
+        result = plan.exec(
+            service_name = foundry_service.name,
+            recipe = ExecRecipe(
+                command = ["/bin/sh", "-c", "cd {} && forge build".format(contract_path)],
+            ),
+        )
+        plan.verify(result["code"], "==", 0)
+
+        script_output = exec_on_service(
+            plan,
+            foundry_service.name,
+            "cd {} && forge script {}:{} --broadcast --rpc-url {} {} --json  --skip test > output.json ".format(
+                contract_path,
+                script_path,
+                contract_name,
+                rpc_url,
+                wallet_command,
+            ),
+        )
 
     exec_on_service(
         plan,
         foundry_service.name,
         "cat {}/output.json ".format(contract_path),
     )
```

Comment:
The idea here is in case of any run - let's sa a shell script for dependency which is taking care of everything internally, it should have this `output.json` file to display it to user.  This would give the output of the scripts which ran internally.